### PR TITLE
Port, address clarifications

### LINKED ISSUE #250: Disconnect between server/client(.socket).port and network.transport (TCP/UDP)
> It seems that the `network.transport` attribute was moved to trace-specific attributes, and the port is ambiguous if we don't know whether it was UDP or TCP

_Originally posted by @Oberon00 in https://github.com/open-telemetry/semantic-conventions/pull/244#discussion_r1289803009_

### PR DESCRIPTION
**Pull request by Oberon00**
Fixes #250

## Changes

* Encourage setting `network.transport` when reporting port numbers
* (No changelog:) Re-arrange section structure of networking-related attributes (keep them in general attributes.md for now)

## Merge requirement checklist

* [X] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [X] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* ~~[ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.~~

### CODE REVIEW

File: model/network.yaml
Comment (joaopgrassi) [inline model/network.yaml:29]: I wonder if such clarification makes also sense on the `port` attributes, such as [server.port](https://github.com/open-telemetry/semantic-conventions/blob/4bbb8c907402caa90bc077214e8a2c78807c1ab9/model/server.yaml#L20C13-L20C13).

Just thinking, if such notice is only present here, when I'm looking at the port attribute I might miss that I should also add the protocol 🤔

Comment (Oberon00) [reply model/network.yaml:29]: > I wonder if such clarification makes also sense on the port attributes, such as [server.port](https://github.com/open-telemetry/semantic-conventions/blob/4bbb8c907402caa90bc077214e8a2c78807c1ab9/model/server.yaml#L20C13-L20C13).

To be honest, I initially wanted to put it there but that would be so much repetition ((server|client|source|destination).port, (server|client).socket.port.

Comment (Oberon00) [reply model/network.yaml:29]: (Keeping it unresolved for now, it's a good thing to note; should not block merging from my perspective and unless others feel strongly about it I will keep it as-is)

File: docs/general/attributes.md
Comment (joaopgrassi) [inline docs/general/attributes.md:54]:
```suggestion
For all IP-based protocols, the "address" should be just the IP-level address.
```
I'm not sure we need the "For the purpose of these semantic conventions" part. Also since this is general/attributes.md, it might be confusing as to what "these semantic conventions" refer to.

File: docs/general/attributes.md
Comment (joaopgrassi) [inline docs/general/attributes.md:55]:
```suggestion
Protocol-specific parts of an address are split into other attributes (when applicable) such as "port" attributes for
```
I would start this in its own sentence, as all together it reads quite long.

File: docs/general/attributes.md
Comment (joaopgrassi) [inline docs/general/attributes.md:57]:
```suggestion
identify the transport, then setting the [`network.transport`](#other-network-attributes) attribute is especially encouraged.
```

Do we want to use any normative wording here? Like :

> identify the transport, then [`network.transport`](#other-network-attributes) SHOULD be set.

Comment (Oberon00) [reply docs/general/attributes.md:57]: Let's instead use this:

```suggestion
identify the transport, then setting [`network.transport`](#other-network-attributes) is especially encouraged.
```

I think for normative wording we would really need to add it (or at least a link) to each single attribute. I don't think there is really a need to make it normative either.

Comment (joaopgrassi) [reply docs/general/attributes.md]: Sounds good :)

**Review by jsuereth — APPROVED** [review]

File: docs/database/database-spans.md
Comment (trask) [inline docs/database/database-spans.md:70]: the markdown table is rendering weird, I think because of the extra newline

**Review by trask — APPROVED** [review]

File: model/network.yaml
Comment (Oberon00) [inline model/network.yaml:29]: Addressing: https://github.com/open-telemetry/semantic-conventions/pull/289/files#r1310991535

```suggestion
          The value SHOULD be normalized to lowercase.
          Consider always setting the transport when setting a port number, since
```

No line breaks allowed in brief, but I also don't want to hide it in a "note"

**Review by joaopgrassi — APPROVED** [review]

File: docs/database/database-spans.md
Comment (Oberon00) [reply docs/database/database-spans.md]: Resolved via https://github.com/open-telemetry/semantic-conventions/pull/289#discussion_r1311213181 and 4a7f3f6edd047e2fb2fa11f2e04442fe3334c5c0

File: docs/http/http-spans.md
Comment (mateuszrzeszutek) [inline docs/http/http-spans.md:107]: The `[4]` footnote says:

> If not default (`tcp` for `HTTP/1.1` and `HTTP/2`, `udp` for `HTTP/3`).

Doesn't that conflict withe the statement you added? If you only have port, set transport; unless you also have the protocol, in which case, don't do that?
Should we perhaps remove the footnote?

Comment (Oberon00) [reply docs/http/http-spans.md:107]: No, not at all. [4] states the cases in which it is required, it is totally OK if there is a recommendation that is stricter than the requirent.

**Review by lmolkova — APPROVED** [review]

File: model/network.yaml
Comment (lmolkova) [inline model/network.yaml:28]: Maybe we can set the requirement level to something `conditionally_required: if instrumented technology supports different transports and value cannot be inferred from other attributes such as `network.protocol.name`` - this would also eliminate the concern for ports and require instrumentations to populate it whenever it's helpful.

Also, it will make brief more brief.

**Comment by Oberon00** [comment]
Is there any blocker for merging this? I resolved the old conversations, IMHO this is ready.

File: model/network.yaml
Comment (trask) [reply model/network.yaml:28]: > Also, it will keep brief more brief.

ya, I think we have a (general) brief problem, tracking here: #335

**Comment by trask** [comment]
> Is there any blocker for merging this? I resolved the old conversations, IMHO this is ready.

@open-telemetry/specs-semconv-maintainers bump

File: model/network.yaml
Comment (trask) [reply model/network.yaml:28]: opened #353 to track https://github.com/open-telemetry/semantic-conventions/pull/289#discussion_r1336505431, resolving this conversation

**Pull request merged**
